app/Importers/BinBinDataImporter.php

```diff
@@ -0,0 +1,69 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\Importers;
+
+use GuzzleHttp\Exception\GuzzleException;
+use Symfony\Component\DomCrawler\Crawler;
+
+class BinBinDataImporter extends DataImporter
+{
+    public const language = "en";
+
+    protected Crawler $sections;
+
+    public function extract(): static
+    {
+        try {
+            $response = $this->client->get("https://www.binbin.tech/lokasyonlar");
+            $html = $response->getBody()->getContents();
+        } catch (GuzzleException) {
+            $this->createImportInfoDetails("400", self::getProviderName());
+
+            $this->stopExecution = true;
+
+            return $this;
+        }
+
+        $crawler = new Crawler($html);
+        $this->sections = $crawler->filter(".location-wrapper .cities-information .div-block-35");
+
+        if (count($this->sections) === 0) {
+            $this->createImportInfoDetails("204", self::getProviderName());
+
+            $this->stopExecution = true;
+        }
+
+        return $this;
+    }
+
+    public function transform(): void
+    {
+        if ($this->stopExecution) {
+            return;
+        }
+
+        $existingCityProviders = [];
+
+        foreach ($this->sections as $section) {
+            $data = explode("|", $section->nodeValue);
+
+            if (trim($data[1]) === "Türkiye") 
+            $countryName = "Turkey"; 
+            else $countryName = $this->translate(trim($data[1]), self::language);
+
+            if (trim($data[0]) === "Uşak") 
+            $cityName = "Usak"; 
+            elseif (trim($data[0]) === "Murter") 
+            continue;
+            else $cityName = $this->translate(trim($data[0]), "en");
+
+            $provider = $this->load($cityName, $countryName);
+
+            if ($provider !== "") $existingCityProviders[] = $provider;
```

Comment:
This code formatting is fucked up. PHPStorm Ctrl+Alt+L should fix it, but you can also wait until this is merged: https://github.com/blumilksoftware/codestyle/pull/99

Reply:
fixed